### PR TITLE
Add e-bulletin signups

### PR DESCRIPTION
**Pull request by mattandrews**
Using the newly-discovered Dotmailer API, this lets us automate signups and reinstate the original form. Hooray!

This tracks successful signups via (anonymous) Google Analytics events, and logs failed ones to Sentry. Thanks @davidrapson for both those suggestions.

I've verified this working on Wired's marketing tool but worth someone else having a go just to double check.

### CODE REVIEW

File: controllers/toplevel/index.js [hunk 1]
```diff
@@ -6,13 +6,17 @@ const rp = require('request-promise');
 const ab = require('express-ab');
 const { sortBy } = require('lodash');
 const xss = require('xss');
+const Raven = require('raven');
+const _ = require('lodash');
```
Comment (davidrapson) [inline controllers/toplevel/index.js]: I don't see lodash being used anywhere in this file?

File: controllers/toplevel/index.js [hunk 2]
```diff
@@ -180,16 +184,14 @@ module.exports = (pages, sectionPath, sectionId) => {
                     res.redirect('/#' + config.get('anchors.ebulletin'));
                 });
             } else {
-                // convert location into proper field value
-                let location = xss(req.body['location']);
-                req.body[location] = 'yes';
-                delete req.body['location'];
-
+                let newsletterLocation = req.body['location'];
```
Comment (davidrapson) [inline controllers/toplevel/index.js]: `req.body.location` instead of `req.body['location']`?

File: controllers/toplevel/index.js [hunk 3]
```diff
@@ -180,16 +184,14 @@ module.exports = (pages, sectionPath, sectionId) => {
...
                 let locale = req.body.locale;
                 let localePrefix = locale === 'cy' ? config.get('i18n.urlPrefix.cy') : '';
 
                 // redirect errors back to the homepage
-                let handleSignupError = () => {
+                let handleSignupError = errMsg => {
+                    // pass error to Sentry
```
Comment (davidrapson) [inline controllers/toplevel/index.js]: This comment feels like it could go

File: controllers/toplevel/index.js (hunk 1 above)
Comment (mattandrews) [reply controllers/toplevel/index.js]: Ah yeah, was using it and forgot to remove it. Normally my IDE alerts me to unused variables, but in this case, the variable name `_` made the error underline quite difficult to spot...
image

**Comment by davidrapson** [comment]
🙏  Working for me, LGTM

**Pull request merged**